### PR TITLE
[feature] Gnome sort

### PR DESCRIPTION
**Pull request by kvedala**
#### Description of Change

This is a continuation of #828 since it seems abandoned. 
The PR implements the Gnome sort algorithm.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/956"></a>

### CODE REVIEW

File: sorting/gnome_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+namespace sorting {
+/**
+ * This implementation is for a C-style array input that gets modified in place.
+ * @param [in,out] arr our array of elements.
+ * @param size size of given array
+ */
+template <typename T>
+void gnomeSort(T *arr, int size) {
+    // few easy cases
+    if (size <= 1) return;
```
Comment (Panquesito7) [inline sorting/gnome_sort.cpp:35]: Please make it look like C++ code:

```suggestion
    if (size <= 1) { return; }
```
Same in other parts of the code.

Comment (kvedala) [reply sorting/gnome_sort.cpp:35]: Please refrain from commenting on pull-requests marked as "draft". thank you

Comment (kvedala) [reply sorting/gnome_sort.cpp]: See what happened with the auto-tools: 8419dbfce126cd241ec20bbe1892c90d9d6667a7

**Comment by kvedala** [comment]
@beqakd Feel free to review as well. Thanks

**Comment by kvedala** [comment]
####Note: 
Apparently, for GitHub Actions to execute check-out, the PR should not be merged. 
In this particular case, the last commit that could not be checked for CI was only doc change while the commit just before it passed all CI checks.

**Pull request merged**